File: content-solutions/two-pointers/rain-garden.java
class Solution {
    public int solution(int[] heights) {
        int l = 0, r = heights.length - 1, lmax = 0, rmax = 0, water = 0;
        while (l <= r) {
            if (lmax <= rmax) {
                lmax = Math.max(lmax, heights[l]);
                water += lmax - heights[l];
                l++;
            } else {
                rmax = Math.max(rmax, heights[r]);
                water += rmax - heights[r];
                r--;
            }
        }
        return water;
    }
}

File: content-solutions/two-pointers/rain-garden.py
def solution(heights):
    l, r = 0, len(heights) - 1
    lmax = rmax = 0
    water = 0
    while l <= r:
        if lmax <= rmax:
            lmax = max(lmax, heights[l])
            water += lmax - heights[l]
            l += 1
        else:
            rmax = max(rmax, heights[r])
            water += rmax - heights[r]
            r -= 1
    return water
